filedescriptor/src/windows.rs

```diff
@@ -208,6 +213,43 @@ impl OwnedHandle {
             })
         }
     }
+
+    pub(crate) fn redirect_stdout_to_file(&self) -> anyhow::Result<Redirection> {
+        let std_original = unsafe{ GetStdHandle(STD_OUTPUT_HANDLE) } ;
+        if unsafe {SetStdHandle(STD_OUTPUT_HANDLE, self.handle)} == 0 {
```

Comment:
I think this should assign a duplicate of `self.handle`, otherwise when `self` is dropped, we break the output for the whole program.

Really, I think this should be a "class" function and not accept `self` but instead be passed something that is `AsRawFileDescriptor`.  Then this logic would be to `FileDescriptor::dup()` the passed handle, and if `SetStdHandle` succeeds, that duplicated handle would have `into_raw_descriptor` called on it to transfer ownership to the OS.

Reply:
Have made this change to the code now